### PR TITLE
tmpfiles: Allow configuration to ignore execution errors

### PR DESCRIPTION
**Pull request by bryteise**
This is an implementation that covers making errors encountered when writing
file content optionally fatal. If this is something that folks would want I'll
add handling of this for all the other directives. I'd appreciate suggestions
on how this might better be structured as well (use of a goto fail or such) as
I'm not super happy with the approach.

### CODE REVIEW

**Comment by poettering** [comment]
This sounds OK to have, but I think eating up the errors in case this is used is better done in the caller, i.e. in process_item() after create_item() returned...

Question is also, should "-" only have an effect on creation? What about removal or clean-up? I am tempted to say only creation failures should be affected by it...

This also needs a man page update.

**Comment by bryteise** [comment]
@poettering Ah yes of course, moving to the caller was the right thing to do.

I think this only makes sense for creation, I couldn't quite come up with a use case where clean or remove would want this without having a different (and better way) to handle the same problem.

Man page change included. Thanks for the review =).

File: src/tmpfiles/tmpfiles.c [hunk 1]
```diff
@@ -2271,6 +2273,10 @@ static int process_item(Item *i) {
         r = arg_create ? create_item(i) : 0;
         q = arg_remove ? remove_item(i) : 0;
         p = arg_clean ? clean_item(i) : 0;
+        /* Failure can only be tolerated for create */
+        if (i->allow_failure) {
+                r = 0;
+        }
```
Comment (poettering) [inline src/tmpfiles/tmpfiles.c]: please follow coding style: no {} around single-line if blocks, please. (See CODING_STYLE for details)

**Comment by poettering** [comment]
(dropped the RFC thing from the subject, as this PR is more than just an RFC now)

**Comment by bryteise** [comment]
Ugh sorry, have the brackets hardcoded in my head from different projects X(. Thanks for the reviews!

**Pull request merged**